service_worker_api/index.d.ts

```diff
@@ -858,23 +721,27 @@ interface ServiceWorkerGlobalScope {
      *
      * @param [messageevent]
      */
-    onmessage: (messageevent?: MessageEvent) => void;
+    // `onmessage` is actually fired with `ExtendableMessageEvent`, but
+    // since we are merging the interface into `Window`, we should
+    // make sure it's compatible with `window.onmessage`
+    // onmessage: (messageevent: ExtendableMessageEvent) => void;
+    onmessage: (messageevent: MessageEvent) => void;
 
     /**
      * An event handler fired whenever a notificationclick event occurs — when
      * a user clicks on a displayed notification.
      *
      * @param [notificationclickevent]
      */
-    onnotificationclick: (notificationclickevent?: NotificationEvent) => void;
+    onnotificationclick: (notificationclickevent: NotificationEvent) => void;
 
     /**
      * An event handler fired whenever a push event occurs — when a server
      * push notification is received.
      *
      * @param [onpushevent]
```

Comment:
I think that is because of the `@param [notificationclickevent]` annotation. If I make the parameter as optional, typescript will give a hint like `(parameter) notificationclickevent: NotificationEvent | undefined`.